src/scanner/mod.rs

```diff
@@ -48,35 +49,40 @@ impl Scanner {
     /// Returns all open ports as Vec<u16>
     pub async fn run(&self) -> Vec<SocketAddr> {
         let ports: Vec<u16> = self.port_strategy.order();
-        let batch_per_ip: usize = self.batch_size as usize / self.ips.len();
+        // let batch_per_ip: usize = self.batch_size as usize / self.ips.len();
         let mut open_sockets: Vec<SocketAddr> = Vec::new();
-
-        for batch in ports.chunks(batch_per_ip) {
-            let mut sockets = self.scan_ports(batch).await;
-            open_sockets.append(&mut sockets);
-        }
-
-        open_sockets
-    }
-
-    /// Given a slice of sockets, scan them all.
-    /// Returns a vector of open sockets.
-    async fn scan_ports(&self, ports: &[u16]) -> Vec<SocketAddr> {
+        let mut targets: VecDeque<SocketAddr> =
+            VecDeque::with_capacity(self.ips.len() * ports.len());
```

Comment:
What does this code do? :)

Reply:
Creates a VecDeque, a double ended queue with capacity. Take the number of ips and ports, multiply them and we have the amount of ip:port combinations go into the targets vector. Reason to use with_capacity is to avoid memory reallocation when the vector have to grow.